### PR TITLE
[hotfix] Add MiniClusterExtension to ITCase tests

### PR DESCRIPTION
**Pull request by hlteoh37**
## Purpose of the change

* [Text for ITCase MiniCluster ArchUnitTest has been changed for Flink 1.18](https://github.com/apache/flink/pull/22399). This PR removes the need for this exception violation in ITCase, by registering a local MiniCluster environment in the Firehose ITCase.
* Change in the github workflow is merely to show that this case passes against `1.18-SNAPSHOT`. It will be removed before merging.

## Verifying this change
This change is to fix breaking ArchUnitTests.

## Significant changes
*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterwards, for convenience.)*
- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [ ] New feature has been introduced
  - If yes, how is this documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by hlteoh37** [comment]
Verified that checks against 1.18-SNAPSHOT pass
image

File: flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java [hunk 1]
```diff
@@ -30,10 +30,6 @@
 import org.apache.flink.connector.kinesis.source.split.KinesisShardSplit;
 import org.apache.flink.connector.kinesis.source.split.StartingPosition;
 
-import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableList;
```
Comment (Samrat002) [inline flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java:33]: Are we getting rid of guava ? 
I dont see any dependency removal of guava in pom file ?

**Review by Samrat002 — COMMENTED** [review]
com.google.common.collect.ImmutableSet and java.util.stream.Stream are two different concepts in Java that serve distinct purposes:

*com.google.common.collect.ImmutableSet:*

ImmutableSet is part of the Google Guava library, which provides various utility classes and data structures. ImmutableSet is used to create a set (a collection that doesn't allow duplicate elements) that cannot be modified after creation. This is particularly useful in scenarios where you want to ensure that the contents of a set remain constant throughout its lifecycle. Once an ImmutableSet is created, you can't add, remove, or modify elements in it.

For example:

```
ImmutableSet<String> colors = ImmutableSet.of("red", "green", "blue");
```

*java.util.stream.Stream:*
Stream is part of the Java Streams API introduced in Java 8. A Stream represents a sequence of elements that can be processed in various ways, such as filtering, mapping, and reducing. Streams allow you to perform operations on a collection of data in a functional and often more concise manner compared to traditional loops. Unlike ImmutableSet, a Stream doesn't hold data itself; it's a pipeline through which data can flow.

For example, you can use a Stream to filter and transform a list of numbers:

```
List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);
Stream<Integer> squaredEvenNumbers = numbers.stream()
    .filter(n -> n % 2 == 0)
    .map(n -> n * n);
```

In summary, ImmutableSet is a specific data structure that provides an immutable set of elements, while Stream is a broader concept that represents a sequence of elements and allows you to perform various operations on that sequence. The choice between them depends on the use case: if you need an immutable set, you'd go for ImmutableSet, and if you need to process and manipulate data in a functional way, you'd use Stream.

Some Code changes here in test cases changes from ImmutableMap/ImmutableSet to stream then wrapping it up with `Collections.unmodifiableMap(`. Isn't that adds extra computations ?

File: flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java (hunk 1 above)
Comment (hlteoh37) [reply flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java:33]: This comes from the shaded guava! We are removing this dependency so that we don't have to keep upgrading this when Flink upgrades shaded guava

Comment (hlteoh37) [reply flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumerator.java:33]: All the changes that changes `ImmutableXXX` to `Streams` were in tests. We do not need to use a `ImmutableXXX`

**Review by Samrat002 — APPROVED** [review]
Thank you for thorough explanation, @hlteoh37 , regarding the core modifications wrt wrapping using Collections + performance and their anticipated minimal/no impact.

I'm in agreement, the changes looks good to me 🚀.
